### ISSUE TITLE
Error: Call to a member function setTimezone() on boolean

### ISSUE THREAD

**Issue by aborderon** [nextcloud/circles#221]
Hello,

I use the Nextcloud version 13.0.5 and the last Circle extension.

I'm having some problems, when I try to add LDAP users to a circle, I get the following error in the Nextcloud logs.

![image](https://user-images.githubusercontent.com/6728728/43899477-5885f2bc-9be2-11e8-8fd8-99a34e87e274.png)

The circle then becomes unusable because I had error 500 in the UI. It is also impossible to delete it.

Thank you.

**Comment by fgsl** [comment]
Hi, @aborderon. Can you tell if table oc_circles_members has some record with a empty column 'joined'?

**Comment by aborderon** [comment]
@fgsl I have my LDAP user in the table **oc_circles_members** and a value in the **joined** column.

![image](https://user-images.githubusercontent.com/6728728/43941779-bb9d74d6-9c75-11e8-84c9-967e212d6573.png)

**Comment by fgsl** [comment]
@aborderon, I would like to ask you to make a little change for logging more details.

Please, in your installation, add before line 236 of /var/www/nextcloud/apps/circles/lib/Db/MembersRequestBuilder.php the following:

\OC::$server->getLogger()->error(__METHOD__ . ' data: '. print_r($data,true));

Try to add a user, according to your initial description, and please report the log message for this method.

Thank you.

**Comment by aborderon** [comment]
Here is the output after adding an LDAP user in a private circle :

`METHOD data: Array ( [circle_id] => ccc43642bf9fd7 [group_id] => Direction [level] => 1 [note] => [joined] => )`

`Use of undefined constant METHOD - assumed 'METHOD' at /var/www/nextcloud/apps/circles/lib/Db/MembersRequestBuilder.php#235`

Once added, I do not see the user in the circle.

For public circles, I feel that it works well.

**Comment by fgsl** [comment]
Great. We know that key 'joined' is empty at this moment.

Now, please, add the following line before  line 67 of /var/www/nextcloud/apps/circles/lib/Db/MembersRequestBuilder.php 

`OC::$server->getLogger()->error(__METHOD__ . ' UTC Date: '. $this->timezoneService->getUTCDate());`

(excuse me, in last message I forgot to quote line and underscores of `__METHOD__` disappeared):

Please, after that change, add a user again and report log message.

Thank you.

**Comment by aborderon** [comment]
Here is the output, I have a date obviously.

`OCA\Circles\Db\MembersRequestBuilder::getMembersInsertSql UTC Date: 2018-08-10 13:36:31`

**Comment by fgsl** [comment]
Well, we know that date is generated and date is saved, now we will check if date is coming back from table.
Please, add the following line before line 68 of /var/www/nextcloud/apps/circles/lib/Db/MembersRequest.php:

`OC::$server->getLogger()->error(__METHOD__ . ' Member data: '. print_r($data,true));`

Paste the output here, please.

**Comment by aborderon** [comment]
there is no $**data** argument defined in the function.

`OCA\Circles\Db\MembersRequestBuilder::getMembersInsertSql Member data:`

`Undefined variable: data at /var/www/nextcloud/apps/circles/lib/Db/MembersRequestBuilder.php#68`

**Comment by aborderon** [comment]
If it helps you, I have this error in the chrome console on the circles list.

![image](https://user-images.githubusercontent.com/6728728/43963479-b4b92a12-9cba-11e8-8a00-f8611f02da0f.png)

**Comment by fgsl** [comment]
Probably error in Chrome console is because absence of joined value. 

Find method forceGetMember and verify if code is equals to [my last comment](https://github.com/nextcloud/circles/issues/221#issuecomment-412094548).

**Comment by aborderon** [comment]
The method is equal but it is in the MembersRequest.php file and not MembersRequestBuilder.php

https://github.com/nextcloud/circles/blob/7c98ce3098308bffe87b953d124e38caddbf5623/lib/Db/MembersRequest.php

**Comment by fgsl** [comment]
Yes, it's MembersRequest.php, [as I said](https://github.com/nextcloud/circles/issues/221#issuecomment-412087021). Then, please, add the line:

`OC::$server->getLogger()->error(__METHOD__ . ' Member data: '. print_r($data,true));`

Before  statement if and report log here.

**Comment by aborderon** [comment]
Oh sorry I did not see the file change ...

I just added the logger but I have nothing in the logs after adding a user.

**Comment by fgsl** [comment]
Debug the code with a breakpoint in item 3 of [your initial log snippet](https://github.com/nextcloud/circles/issues/221#issue-349111861). Try to discover more details in reading of database.

**Comment by fgsl** [comment]
Locate this method in file MembersRequest.php:

```
	public function forceGetHigherLevelGroupFromUser($circleUniqueId, $userId) {
		$qb = $this->getGroupsSelectSql();

		$this->limitToCircleId($qb, $circleUniqueId);
		$this->limitToNCGroupUser($qb, $userId);

		/** @var Member $group */
		$group = null;

		$cursor = $qb->execute();
		while ($data = $cursor->fetch()) {
			$entry = $this->parseGroupsSelectSql($data);
			if ($group === null || $entry->getLevel() > $group->getLevel()) {
				$group = $entry;
			}
		}
		$cursor->closeCursor();

		return $group;
	}
```

And add one line inside while loop for method to be so:

```
	public function forceGetHigherLevelGroupFromUser($circleUniqueId, $userId) {
		$qb = $this->getGroupsSelectSql();

		$this->limitToCircleId($qb, $circleUniqueId);
		$this->limitToNCGroupUser($qb, $userId);

		/** @var Member $group */
		$group = null;

		$cursor = $qb->execute();
		while ($data = $cursor->fetch()) {
                        // for logging member data and to see value of column joined
                        OC::$server->getLogger()->error(__METHOD__ . ' Member data: '. print_r($data,true));
			$entry = $this->parseGroupsSelectSql($data);
			if ($group === null || $entry->getLevel() > $group->getLevel()) {
				$group = $entry;
			}
		}
		$cursor->closeCursor();

		return $group;
	}
```

Tell us the results.

**Comment by aborderon** [comment]
I tried well but I do not fit in the loop. **$cursor-> fetch()** does not return a result.

I put a logger just before the loop to analyze the query: 

`\OC::$server->getLogger()->error(__METHOD__ . ' Query: '. print_r($cursor,true));`

Here is the output :
`OCA\Circles\Db\MembersRequest::forceGetHigherLevelGroupFromUser Query: Doctrine\DBAL\Driver\PDOStatement Object ( [queryString] => SELECT `g`.`circle_id`, `g`.`group_id`, `g`.`level`, `g`.`note`, `g`.`joined` FROM `oc_circles_groups` `g`, `oc_group_user` `ncgu` WHERE (`g`.`circle_id` = ?) AND ((`g`.`group_id` = `ncgu`.`gid`) AND (`ncgu`.`uid` = ?)) )`

**Comment by fgsl** [comment]
Let me to understand better.

When you said that **$cursor->fetch()** does not return a result, which exactly is the type returned?

Because, if type is something considered false, it does not call method **parseGroupsSelectSql**, but your initial error trace shows that **parseGroupsSelectSql** is called.

**Comment by daita** [comment]
Hello,

This fix should keep the error message from the title of this report away.

About your circles you cannot delete, do you have more stuff in the server logs, or in the Javascript console of your browser ?

**Comment by aborderon** [comment]
Hello,
no I have nothing more except the previous items ...

**Comment by joshtrichards** [comment]
Main issue fixed in 5824dca.
Deletion matter unknown. Hopefully that got worked out. Given it's been 6 years I suspect it's either fixed or something else was going on here. I'll close this, but feel free to nudge me if you want to revisit this ancient issue. We can re-open it with updated info (or probably better given time: open a new issue with up-to-date info).
